oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -191,9 +191,14 @@ public FileSystem getFileSystem() {
     public OSProcess[] getProcesses(int limit, ProcessSort sort) {
         List<OSProcess> procs = new ArrayList<>();
         File[] pids = ProcUtil.getPidFiles();
+        
+        LinuxUserInfo userInfo = getUserInfo();

```

Comment:
Calling the code this way will re-parse the user and group list every time.  However, this information rarely changes in reality.  Is there a way to check when this information is last updated (e.g., timestamp on `/etc/shadow` (if available) or `/etc/passwd`)?  If so, this information could be stored/cached as a class variable and then updated only if the timestamp is changed.   

Alternately you could still cache and trigger a re-read upon an error (finding a UID not in the map) or after a longer time period (e.g., 5 minutes).

Reply:
I like the idea of caching. I was thinking about it when I implemented but I don't know how to find out when the cache gets dirty. However, I think making it dirty after a few minutes is an ok approach for a users/group database that changes rarely and anyways sometimes needs some time to propagate (NIS setup with cached information). I will change it to this approach.

Reply:
It's easy to find a dirty cache with a new user -- they won't be in the cache when you look up by UID.  Simply refresh and you're done.  As for a deleted user -- doesn't matter if the extra user hangs around in the cache if you're never looking them up.   The only issue you'll have is a renamed user (is that even possible?) and I think a few-minute cache won't hurt there.